docs/use-cases/Intel-Discrete-GPU-passthrough-and-Kata.md

```diff
@@ -0,0 +1,255 @@
+
+Some hardware vendors use a different name in BIOS, such as:
+
+- Above 4G Decoding
+- Memory Hole for PCI MMIO
+- Memory Mapped I/O above 4GB
+
+## Host Kernel Requirements
+
+For device passthrough to work with the Max and Flex Series, an out of tree kernel driver is required.
+
+For Ubuntu 22.04 server, follow these instructions to install the out of tree GPU driver:
+```
+$ sudo apt update
+$ sudo apt install -y gpg-agent wget
+$ wget -qO - https://repositories.intel.com/gpu/intel-graphics.key | \
+    sudo gpg --dearmor --output /usr/share/keyrings/intel-graphics.gpg
+$ echo "deb [arch=amd64 signed-by=/usr/share/keyrings/intel-graphics.gpg] https://repositories.intel.com/gpu/ubuntu ${VERSION_CODENAME}/lts/2350 unified" | \
+    sudo tee /etc/apt/sources.list.d/intel-gpu-${VERSION_CODENAME}.list
+$ sudo apt update
+$ sudo apt install -y linux-headers-$(uname -r) flex bison intel-fw-gpu intel-i915-dkms xpu-smi
+```
+
+For support on other distributions, please refer to [DGPU-docs](https://dgpu-docs.intel.com/driver/installation.html)
+
+Additionally, verify that the following kernel configs are enabled on your host kernel:
+```
+CONFIG_VFIO
+CONFIG_VFIO_IOMMU_TYPE1
+CONFIG_VFIO_PCI
+```
+
+## Host kernel command line 
+
+Your host kernel needs to be booted with `intel_iommu=on` and `i915.enable_iaf=0` on the kernel command
+line.
+
+1. Run the following to change the kernel command line using grub
+```
+sudo vim /etc/default/grub
+```
+
+2. At the end of the GRUB_CMDLINE_LINUX_DEFAULT append the below line:
+
+`intel_iommu=on iommu=pt i915.max_vfs=63 i915.enable_iaf=0`
+
+3. Update grub as per OS distribution:
+
+For ubuntu:
+```
+sudo update-grub
+```
+
+For Centos/RHEL:
+```
+sudo grub2-mkconfig -o /boot/grub2/grub.cfg 
+``` 
+
+4. Reboot the system
+```
+sudo reboot
+```
+
+## Install and configure Kata Containers
+
+To use this feature, you need Kata version 1.3.0 or above.
+Follow the [Kata Containers setup instructions](../install/README.md)
+to install the latest version of Kata.
+
+To use large BARs devices (for example, NVIDIA Tesla P100), you need Kata version 1.11.0 or above.
+
+In order to pass a GPU to a Kata Container, you need to enable the `hotplug_vfio_on_root_bus`
+configuration in the Kata `configuration.toml` file as shown below.
+
+```
+$ sudo sed -i -e 's/^# *\(hotplug_vfio_on_root_bus\).*=.*$/\1 = true/g' /usr/share/defaults/kata-containers/configuration.toml
+```
+
+Make sure you are using the `q35` machine type by verifying `machine_type = "q35"` is
+set in the `configuration.toml`. Make sure `pcie_root_port` is set to a positive value.
+
+After making the above changes, configuration in the `configuration.toml` should look like this:
+```
+machine_type = "q35"
+
+hotplug_vfio_on_root_bus = true
+pcie_root_port = 1
+```
+
+## GPU passthrough with Kata Containers
+
+Use the following steps to pass an Intel discrete GPU  with Kata:
+
+1. Find the Bus-Device-Function (BDF) for GPU device:
```

Comment:
How is the VF enablement done?

Reply:
@ mythi the command mentioned earlier `echo 4 | sudo tee  /sys/bus/pci/devices/0000\:3a\:00.0/sriov_numvfs` creates the VFs. The out of tree kernel driver along with the kernel command line mentioned enables the VFs to be created.

Reply:
Are the VFs automatically bound to `vfio-pci`?